src/lib/server/generateFromDefaultEndpoint.ts

```diff
@@ -21,24 +21,76 @@ export async function generateFromDefaultEndpoint(
 		return_full_text: false,
 	};
 
-	const endpoint = modelEndpoint(defaultModel);
-	let { generated_text } = await textGeneration(
-		{
-			model: endpoint.url,
+	const randomEndpoint = modelEndpoint(defaultModel);
+
+	const abortController = new AbortController();
+
+	let resp: Response;
+
+	if (randomEndpoint.host === "sagemaker") {
+		const requestParams = JSON.stringify({
+			...newParameters,
 			inputs: prompt,
-			parameters: newParameters,
-		},
-		{
-			fetch: (url, options) =>
-				fetch(url, {
-					...options,
-					headers: { ...options?.headers, Authorization: endpoint.authorization },
-				}),
-		}
-	);
+		});
+
+		const aws = new AwsClient({
+			accessKeyId: randomEndpoint.accessKey,
+			secretAccessKey: randomEndpoint.secretKey,
+			sessionToken: randomEndpoint.sessionToken,
+			service: "sagemaker",
+		});
```

Comment:
Unfortunately each model can have multiple endpoints (for load balancing) so in that case we can't really create it outside because different endpoints might have different authentication settings.

Reply:
But isn't that `aws` client generic? there is not connection to the URL of the model?

Reply:
When the `AwsClient` gets constructed it needs `randomEndpoint.accessKey` the secret key, etc. 
Those access keys are defined at the endpoint level so if you have two endpoints you'd need to specify it twice like for example:

```
MODELS=`[{
     // bunch of model stuff here
    "endpoints": [
    {
      "host" : "sagemaker",
      "url": "https://runtime.sagemaker.us-east-1.amazonaws.com/****",
      "accessKey": "aaa",
      "secretKey" : "aaa",
      "sessionToken": "aaa",
      "weight": 1
    },
    {
      "host" : "sagemaker",
      "url": "https://runtime.sagemaker.eu-central-1.amazonaws.com/****",
      "accessKey": "bbb",
      "secretKey" : "bbb",
      "sessionToken": "bbb",
      "weight": 5
    },
   {
    "host" : "tgi",
    "url" : "https://api-inference.huggingface.co/*****",
    "authorization" : "Bearer hf_*****",
    "weight" : 3,
    }
    ]
  }
]`
```

In a situation like this the endpoint is determined based on the weight for load balancing and there's no way of knowing which `AwsClient` to use (or if one is needed in such a case). Not sure if that makes sense ?

Reply:
got it thank you!